addon/components/frost-table-header.js

```diff
@@ -95,57 +99,71 @@ export default Component.extend({
-    const lastCategoryIndex = this.accountForSelectionColumn(_categoryColumns.length)
+    const lastCategoryIndex = _categoryColumns.length
     this.$('.frost-table-header-cell').slice(0, lastCategoryIndex)
       .wrapAll(`<${rowTagName} class='${_categoryRowClass} frost-table-row frost-table-header-row'></${rowTagName}>`)
     this.$('.frost-table-header-cell').slice(lastCategoryIndex)
       .wrapAll(`<${rowTagName} class='${_columnRowClass} frost-table-row frost-table-header-row'></${rowTagName}>`)
   },
 
+  alignColumns () {
+    const rowClass = this.get('_hasCategories') ? `.${this.get('_columnRowClass')}` : ''
+    const columnSelector = `${rowClass} .frost-table-cell`
+
+    this.$(columnSelector).toArray().forEach((el) => {
+      const flexBasis = this.parseFloatOrDefault(this.$(el).css('flex-basis'), this.$(el).outerWidth(true))
+      this.$(el).css('flex', `1 0 ${flexBasis}px`)
+    })
+  },
+
   alignCategories () {
-    const {_categoryColumns, _categoryRowClass, _columnRowClass, cellTagName} =
-      this.getProperties('cellTagName', '_categoryColumns', '_categoryRowClass', '_columnRowClass')
+    const {_categoryColumns, _categoryRowClass, _columnRowClass} =
+      this.getProperties('_categoryColumns', '_categoryRowClass', '_columnRowClass')
     const categorySelector = `.${_categoryRowClass} .frost-table-cell`
     const columnSelector = `.${_columnRowClass} .frost-table-cell`
 
-    if (cellTagName === 'th' || cellTagName === 'td') {
-      // Make use of colspan property
-      _categoryColumns.forEach((category, index) => {
-        index = this.accountForSelectionColumn(index)
-        this.$(categorySelector).eq(index).attr('colspan', category.span)
+    let startColumn = 0
+    _categoryColumns.forEach((category, index) => {
+      let categoryFlexBasis = 0
+      let categoryFlexGrow = 0
+      let categoryFlexShrink = 0
+      this.$(columnSelector).slice(startColumn, startColumn + category.span).toArray().forEach((el) => {
+        const flexBasis = parseFloat(this.$(el).css('flex-basis'))
+        const flexGrow = parseFloat(this.$(el).css('flex-grow'))
+        const flexShrink = parseFloat(this.$(el).css('flex-shrink'))
+
+        categoryFlexBasis += flexBasis
+        categoryFlexGrow += flexGrow
+        categoryFlexShrink += flexShrink
       })
-    } else {
-      // Need to determine and set width
-      let startColumn = this.accountForSelectionColumn(0)
-      _categoryColumns.forEach((category, index) => {
-        let totalWidth = 0
-        this.$(columnSelector).slice(startColumn, startColumn + category.span).toArray().forEach((el) => {
-          totalWidth += this.$(el).outerWidth()
-        })
-        this.$(categorySelector).eq(index).css({
-          'width': `${totalWidth}px`
-        })
-        startColumn += category.span
+      this.$(categorySelector).eq(index).css({
+        'flex': `${categoryFlexGrow} ${categoryFlexShrink} ${categoryFlexBasis}px`
```

Comment:
Any reason not to go with the `flex-shrink` and `flex-basis` properties instead of using string template for the shorthand?